logprep/abc/processor.py

```diff
@@ -8,22 +9,20 @@
 
 from attr import define, field, validators
 
-
 from logprep.framework.rule_tree.rule_tree import RuleTree
+from logprep.metrics.metric import Metric, calculate_new_average
 from logprep.processor.base.rule import Rule
-
 from logprep.processor.processor_strategy import SpecificGenericProcessStrategy
 from logprep.util.helper import camel_to_snake
 from logprep.util.json_handling import list_json_files_in_directory
-from logprep.util.processor_stats import ProcessorStats
 from logprep.util.time_measurement import TimeMeasurement
 from logprep.util.validators import file_validator, list_of_dirs_validator
 
 
 class Processor(ABC):
     """Abstract Processor Class to define the Interface"""
 
-    @define(kw_only=True)
+    @define(kw_only=True, slots=False)
```

Comment:
I had to deactivate slots because I am adding dynamically new fields to the config object, namely the metric labels. There were two alternative options (a) letting the end user themselves define all the metric labels via the configuration yaml or (b) changing the processor factory and adjusting all processor initializations to pass the metric labels through. Because they are both far from ideal I decided to remove the slots feature here and adding the metric labels dynamically to the configuration object.